In [17]:
import pandas as pd
import numpy as np
import os

# Trajectory

In [50]:
# Read excel file 
# df = pd.read_excel(r'D:\#DURI-\1. Data\Trajectory.xlsx', sheet_name=0)

# Read from raw data
df = pd.read_csv(r'D:\thomas\project\WELL_TRAJECTORY_DURI.csv')

df.head(20)

UWI FIELD_NAME WELL_NAME SURVEY_SOURCE      MD  AZIMUTH  \
0   DURI00001       DURI    5R-74A      Combined     0.0   359.97   
1   DURI00001       DURI    5R-74A      Combined   899.0   359.97   
2   DURI00002       DURI    4R-76A      Combined     0.0   359.97   
3   DURI00002       DURI    4R-76A      Combined  3049.5   359.97   
4   DURI00003       DURI    4S-42A      Combined     0.0   359.97   
5   DURI00003       DURI    4S-42A      Combined   755.0   359.97   
6   DURI00004       DURI    3K-21A      Combined     0.0   359.97   
7   DURI00004       DURI    3K-21A      Combined   754.0   359.97   
8   DURI00005       DURI    3T-59A      Combined     0.0   359.97   
9   DURI00005       DURI    3T-59A      Combined   778.0   359.97   
10  DURI00006       DURI    4T-31A      Combined     0.0   359.97   
11  DURI00006       DURI    4T-31A      Combined   654.5   359.97   
12  DURI00007       DURI    2K-45A      Combined     0.0   359.97   
13  DURI00007       DURI    2K-45A      Combined  1474.0   359.97   
14  DURI00008       DURI    5P-75A      Combined     0.0   359.97   
15  DURI00008       DURI    5P-75A      Combined  2328.0   359.97   
16  DURI00009       DURI    6M-71A      Combined     0.0   359.97   
17  DURI00009       DURI    6M-71A      Combined  1135.0   359.97   
18  DURI00011       DURI    5Q-83A      Combined     0.0   359.97   
19  DURI00011       DURI    5Q-83A      Combined   676.0   359.97   

    INCLINATION  X_OFFSET      Y_OFFSET         LON       LAT     TVD  
0           0.0       0.0 -2.220450e-16  101.232355  1.295577     0.0  
1           0.0       0.0 -2.220450e-16  101.232355  1.295577   899.0  
2           0.0       0.0  0.000000e+00  101.223728  1.295472     0.0  
3           0.0       0.0  0.000000e+00  101.223728  1.295472  3049.5  
4           0.0       0.0  0.000000e+00  101.220387  1.290059     0.0  
5           0.0       0.0  0.000000e+00  101.220387  1.290059   755.0  
6           0.0       0.0  0.000000e+00  101.208445  1.352268     0.0  
7           0.0       0.0  0.000000e+00  101.208445  1.352268   754.0  
8           0.0       0.0  2.220450e-16  101.216955  1.281500     0.0  
9           0.0       0.0  2.220450e-16  101.216955  1.281500   778.0  
10          0.0       0.0  0.000000e+00  101.217842  1.283046     0.0  
11          0.0       0.0  0.000000e+00  101.217842  1.283046   654.5  
12          0.0       0.0  0.000000e+00  101.202428  1.349074     0.0  
13          0.0       0.0  0.000000e+00  101.202428  1.349074  1474.0  
14          0.0       0.0  0.000000e+00  101.233832  1.312836     0.0  
15          0.0       0.0  0.000000e+00  101.233832  1.312836  2328.0  
16          0.0       0.0  0.000000e+00  101.239832  1.328727     0.0  
17          0.0       0.0  0.000000e+00  101.239832  1.328727  1135.0  
18          0.0       0.0  2.220450e-16  101.232435  1.303210     0.0  
19          0.0       0.0  2.220450e-16  101.232435  1.303210   676.0

In [51]:
wellname = list(set(df['UWI']))
# wellname = [x for x in wellname if x != np.nan]
# wellname = wellname[1:]
print(wellname[:5])

['DURI01314', 'DURI00671', 'DURI00429', 'DURI06320', 'DURI02809']


In [53]:
for well in wellname:
#     per_well = df[df['UWI']==well].loc['UWI', 'MD', 'AZIMUTH', 'INCLINATION']
    per_well = df[['UWI', 'MD', 'AZIMUTH', 'INCLINATION']].loc[df['UWI'] == well]
    per_well.to_csv(r'D:\thomas\project\duri\{}.csv'.format(well), index=False)

## Check number of deviated/horizontal wells
Based on `SURVEY_SOURCE`

In [120]:
survey_combined = df[df.SURVEY_SOURCE == 'Combined']

# total of vertical wells
len(list(set(survey_combined.UWI)))

7054

In [121]:
survey_cpi = df[df.SURVEY_SOURCE == 'CPI']

# total of dev/hor wells CPI
len(list(set(survey_cpi.UWI)))

273

In [122]:
survey_pse = df[df.SURVEY_SOURCE == 'PSE']

# total of dev/hor wells PSE
len(list(set(survey_pse.UWI)))

233

In [125]:
# Deviated/Horizontal wells
len(list(set(survey_cpi.UWI))) + len(list(set(survey_pse.UWI)))

506

In [126]:
# All wells
len(list(set(survey_combined.UWI)))+len(list(set(survey_cpi.UWI)))+len(list(set(survey_pse.UWI)))

7560

Based on `UWI`

In [133]:
df_hor = df.loc[df.UWI.str.contains('H1', na=False)]
len(list(set(df_hor.UWI)))

196

In [134]:
df_dir = df.loc[df.UWI.str.contains('D1', na=False)]
len(list(set(df_dir.UWI)))

310

In [135]:
len(list(set(df_hor.UWI)))+len(list(set(df_dir.UWI)))

506

# Well info

In [58]:
well_info =  pd.read_excel(r'D:\#DURI-\1. Data\Well Header info.xlsx', sheet_name=1, skiprows=1)
well_info.head()

Unnamed: 0  Unnamed: 1  View       Name UWI        Folder Well symbol  \
0         NaN           1  True  DURI00001   1  'Wells/DURI'         Oil   
1         NaN           2  True  DURI00002   2  'Wells/DURI'         Oil   
2         NaN           3  True  DURI00003   3  'Wells/DURI'         Oil   
3         NaN           4  True  DURI00004   4  'Wells/DURI'         Oil   
4         NaN           5  True  DURI00005   5  'Wells/DURI'         Oil   

   Surface X  Surface Y      Latitude    ... Squeeze Squeeze.1  Squeeze.2  \
0  748406.21  143310.32  1°17'44.0913"N  ...     NaN       NaN        NaN   
1  747445.80  143297.90  1°17'43.7145"N  ...     NaN       NaN        NaN   
2  747074.40  142698.80  1°17'24.2271"N  ...     NaN       NaN        NaN   
3  745738.74  149578.83   1°21'8.1810"N  ...     NaN       NaN        NaN   
4  746693.10  141751.80  1°16'53.4172"N  ...     NaN       NaN        NaN   

   Squeeze.3  Squeeze.4  Squeeze.5  Squeeze.6  Original CRS  Created from  \
0        NaN        NaN        NaN        NaN           NaN           NaN   
1        NaN        NaN        NaN        NaN           NaN           NaN   
2        NaN        NaN        NaN        NaN           NaN           NaN   
3        NaN        NaN        NaN        NaN           NaN           NaN   
4        NaN        NaN        NaN        NaN           NaN           NaN   

           Last edited  
0  22/02/2021 21:16:25  
1  22/02/2021 21:16:25  
2  22/02/2021 21:16:25  
3  22/02/2021 21:16:25  
4  22/02/2021 21:16:25  

[5 rows x 252 columns]

In [62]:
well_info = well_info.iloc[:,3:15]

In [124]:
len(well_info)

10765

In [99]:
well_info.sample(15)

Name      UWI        Folder Well symbol  Surface X  Surface Y  \
270      DURI00271      271  'Wells/DURI'         Oil  749414.40  143307.20   
4422     DURI04423     4423  'Wells/DURI'         Oil  748389.86  139789.64   
1105     DURI01106     1106  'Wells/DURI'         Oil  747912.30  141677.20   
10584  DURI10600D1  10600D1  'Wells/DURI'         Oil  745661.70  152158.77   
2707     DURI02708     2708  'Wells/DURI'         Oil  749265.40  143425.50   
2919     DURI02920     2920  'Wells/DURI'         Oil  750017.80  144802.10   
6610     DURI06612     6612  'Wells/DURI'         Oil  747248.68  149060.36   
10239  DURI10254D1  10254D1  'Wells/DURI'         Oil  749155.13  151654.03   
5836     DURI05837     5837  'Wells/DURI'         Oil  747882.15  148686.77   
2513     DURI02514     2514  'Wells/DURI'         Oil  751470.90  145295.00   
3978     DURI03979     3979  'Wells/DURI'         Oil  749637.10  148186.10   
9809     DURI09824     9824  'Wells/DURI'         Oil  749131.18  152077.17   
2260     DURI02261     2261  'Wells/DURI'         Oil  751336.10  145528.80   
10060    DURI10075    10075  'Wells/DURI'         Oil  745198.60  147661.70   
9800     DURI09815     9815  'Wells/DURI'         Oil  748959.78  152708.68   

           Latitude         Longitude   Well datum name  Well datum value  \
270    1°17'43.9607"N  101°14'29.1060"E              KB              67.2   
4422   1°15'49.5102"N  101°13'55.8778"E              KB              83.8   
1105   1°16'50.9549"N  101°13'40.4884"E              KB              68.1   
10584  1°22'32.1495"N  101°12'28.0161"E              KB              53.0   
2707   1°17'47.8151"N  101°14'24.2914"E              KB              60.2   
2919   1°18'32.5948"N  101°14'48.6604"E              KB              61.2   
6610   1°20'51.2623"N  101°13'19.2422"E              KB              53.0   
10239  1°22'15.6168"N  101°14'20.9679"E              KB              65.0   
5836   1°20'39.0849"N  101°13'39.7155"E              KB              57.0   
2513   1°18'48.5935"N  101°15'35.6615"E              KB              65.5   
3978   1°20'22.7382"N  101°14'36.4493"E              KB              61.3   
9809   1°22'29.3886"N  101°14'20.2063"E              KB              64.9   
2260   1°18'56.2064"N  101°15'31.3095"E              KB              66.2   
10060   1°20'5.8023"N  101°12'12.9083"E              KB              65.0   
9800   1°22'49.9464"N  101°14'14.6829"E              KB              66.7   

       TD (TVDSS)  TD (MD)  
270         642.8    710.0  
4422        740.8    824.5  
1105        491.9    560.0  
10584       749.4    802.4  
2707        774.3    834.5  
2919        755.8    817.0  
6610        744.0    797.0  
10239       609.0    674.0  
5836        423.0    480.0  
2513        747.0    812.5  
3978        722.7    784.0  
9809        648.1    713.0  
2260        731.8    798.0  
10060      1400.0   1465.0  
9800        684.3    751.0

In [82]:
horizontal = 'H1' in well_info.UWI
directional = 'D1' in well_info.UWI

In [107]:
hor_wells = well_info.loc[well_info.UWI.str.contains('H1', na=False)]

In [108]:
dir_wells = well_info.loc[well_info.UWI.str.contains('D1', na=False)]

In [113]:
hor_wells.loc[hor_wells.UWI == '10242H1ST1']

Name         UWI        Folder Well symbol  Surface X  \
10227  DURI10242H1ST1  10242H1ST1  'Wells/DURI'         Oil   746932.2   

       Surface Y      Latitude        Longitude   Well datum name  \
10227   147850.4  1°20'11.8929"N  101°13'8.9728"E              KB   

       Well datum value  TD (TVDSS)  TD (MD)  
10227              63.3      2023.7   2087.0

In [127]:
len(hor_wells)

195

In [128]:
len(dir_wells)

308

In [111]:
# totals of deviated and horizontal wells
len(hor_wells) + len(dir_wells)

503

# Check difference

In [147]:
set_df = set(df.UWI)
set_info = set(well_info.Name)

In [148]:
len(set_df)

7560

In [149]:
len(set_info)

10765

In [150]:
set_df.difference(set_info)

{'DURI06978',
 'DURI06979',
 'DURI06980',
 'DURI06981',
 'DURI06982',
 'DURI06983',
 'DURI09464H1',
 'DURI10410D1',
 'DURI10484D1'}

In [155]:
set_info.difference(set_df)

{'DURI09118',
 'DURI06631',
 'DURI06106',
 'DURI09931',
 'DURI04819',
 'DURI06035',
 'DURI10193',
 'DURI10012',
 'DURI09045',
 'DURI10051',
 'DURI10282',
 'DURI09490',
 'DURI09634',
 'DURI09968',
 'DURI10632',
 'DURI09796',
 'DURI09793',
 'DURI08440',
 'DURI08579',
 'DURI07018',
 'DURI09695',
 'DURI04839',
 'DURI05454',
 'DURI08647',
 'DURI08569',
 'DURI06122',
 'DURI09992',
 'DURI06665',
 'DURI09239',
 'DURI09141',
 'DURI09164',
 'DURI09795',
 'DURI09410',
 'DURI08687',
 'DURI08957',
 'DURI06774',
 'DURI06026',
 'DURI09420',
 'DURI06005',
 'DURI06850',
 'DURI05525',
 'DURI09241',
 'DURI10602',
 'DURI10497',
 'DURI05686',
 'DURI09927',
 'DURI08458',
 'DURI10689',
 'DURI08436',
 'DURI08812',
 'DURI06468',
 'DURI07019',
 'DURI05983',
 'DURI04830',
 'DURI09280',
 'DURI09486',
 'DURI10664',
 'DURI04918',
 'DURI05081',
 'DURI05060',
 'DURI09608',
 'DURI09598',
 'DURI05298',
 'DURI06637',
 'DURI09775',
 'DURI05187',
 'DURI05411',
 'DURI10339',
 'DURI08621',
 'DURI08686',
 'DURI06939',
 'DURI

In [161]:
info_diff = list(set_info.difference(set_df))

In [164]:
len(info_diff)

3214

In [163]:
no_traj = [x for x in info_diff if 'D1' in x]
no_traj

[]

Trajectory data from trajectory file : 506 wells<br>
Deviated/hosrizontal wells from well header : 503 wells<br>
Trajectory data without well header : 8 wells (`DURI10410D1` is `DURI10410D1ST1` in well header)